packages/shared/src/session/initializeSession.ts

```diff
@@ -0,0 +1,27 @@
+import { STATUS_RESOLVED } from "../async/createDeferred";
+import { getAuthInfo } from "../authentication/getAuthInfo";
+import { AuthInfo } from "../authentication/types";
+import { deferredAuthInfo, deferredPackageInfo } from "./deferred";
+
+export async function initializeSession({
+  accessToken,
+  packageName,
+  packageVersion,
+}: {
+  accessToken: string | undefined;
+  packageName: string;
+  packageVersion: string;
+}) {
+  if (deferredPackageInfo.status === STATUS_RESOLVED) {
+    return;
+  }
+
+  deferredPackageInfo.resolve({ packageName, packageVersion });
+
+  let authInfo: AuthInfo | undefined = undefined;
+  if (accessToken) {
+    authInfo = await getAuthInfo(accessToken);
```

Comment:
this can throw - what should be the behavior when that happens? a lot of the callers to `initializeSession` create an orphaned promise and it might take the whole pricess down in unexpected ways

Reply:
Good catch. I think for now– log an error (but swallow it)